test/framework/src/main/java/org/elasticsearch/test/InternalAggregationTestCase.java

```diff
@@ -317,13 +325,28 @@ public final void testFromXContent() throws IOException {
         } else {
             originalBytes = toXContent(aggregation, xContentType, params, humanReadable);
         }
+        BytesReference mutated;
+        if (addRandomFields) {
+            /**
+             * - we don't add to the root object because it should only contain the named aggregation to test
```

Comment:
We are not really testing the "aggregations" part of the SearchResponse as a while here. But yes, the current assumption would be that we will not have any other fields in the top-level "aggregations" in the future. Otherwise we need to change Aggregations.fromXContent, which is not tested here.